### PR TITLE
Web3 conn upgrade avirup

### PR DESCRIPTION
**Pull request by avirup2512**
I have added a success message for color of the message (Connection are re established).
I have edit the function called  

- _handleWsSubmit
- _handleRPCChange
- _handleWsChange
in lib/components/NodeControl/index.js file
for creating dynamic ws connection.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-'use babel'
+'use babel';
```
Comment (0mkara) [inline index.js]: must not have semicolon after `'use babel'`. Check other files.

File: lib/components/CoinbaseView/index.js [hunk 2]
```diff
@@ -18,15 +18,16 @@ import React from 'react';
...
 class CoinbaseView extends React.Component {
     constructor(props) {
         super(props);
         this.helpers = props.helpers;
-        this.web3 = props.web3;
+        this.web3 = getWeb3Conn();
         this.state = {
             coinbase: props.accounts[0],
-            balance: 0.00,
+            balance: 0.0,
```
Comment (0mkara) [inline lib/components/CoinbaseView/index.js]: why change `0.00` to `0.0` ?

File: lib/components/CoinbaseView/index.js [hunk 3]
```diff
@@ -40,10 +41,19 @@ class CoinbaseView extends React.Component {
     }
     async componentDidMount() {
         const { coinbase } = this.state;
+        console.log(coinbase);
```
Comment (0mkara) [inline lib/components/CoinbaseView/index.js]: remove unnecessary logs.

File: lib/components/CoinbaseView/index.js [hunk 4]
```diff
@@ -90,32 +100,45 @@ class CoinbaseView extends React.Component {
...
-                            accounts.map((account, i) => {
+                {accounts.length > 0 && (
+                    <div className="row">
+                        <div
+                            className="icon icon-link btn copy-btn btn-success"
+                            onClick={this._linkClick}
+                        />
+                        <select
+                            onChange={this._handleAccChange}
+                            value={this.state.coinbase}
```
Comment (0mkara) [inline lib/components/CoinbaseView/index.js]: use following
```js
const { coinbase } = this.state
...
value={coinbase}
```

File: lib/components/CoinbaseView/index.js [hunk 5]
```diff
@@ -90,32 +100,45 @@ class CoinbaseView extends React.Component {
...
-                    <input
-                        type="submit"
-                        className={this.state.unlock_style}
-                        value="Unlock"
-                    />
-                </form>
+                            })}
+                        </select>
+                        <button onClick={this._refreshBal} className="btn">
+                            {balance} ETH
```
Comment (0mkara) [inline lib/components/CoinbaseView/index.js]: why newline for this ?

File: lib/components/CoinbaseView/index.js [hunk 6]
```diff
@@ -90,32 +100,45 @@ class CoinbaseView extends React.Component {
...
-                        value={password}
-                        onChange={this._handlePasswordChange}
-                    />
-                    <input
-                        type="submit"
-                        className={this.state.unlock_style}
-                        value="Unlock"
-                    />
-                </form>
+                            })}
```
Comment (0mkara) [inline lib/components/CoinbaseView/index.js]: why close all of then in one line ?

File: lib/components/CoinbaseView/index.js [hunk 7]
```diff
@@ -90,32 +100,45 @@ class CoinbaseView extends React.Component {
...
+                        <div className="icon icon-lock" />
+                        <input
+                            type="password"
+                            placeholder="Password"
+                            value={password}
+                            onChange={this._handlePasswordChange}
+                        />
+                        <input
+                            type="submit"
+                            className={this.state.unlock_style}
```
Comment (0mkara) [inline lib/components/CoinbaseView/index.js]: Use spread operator like above.

File: lib/components/NodeControl/index.js [hunk 8]
```diff
@@ -16,30 +16,33 @@
...
+            wsProvider: Object.is(web3.currentProvider.constructor, Web3.providers.WebsocketProvider),
+            httpProvider: Object.is(web3.currentProvider.constructor, Web3.providers.HttpProvider),
             connected: props.web3.currentProvider.connected,
             toAddress: '',
             amount: 0,
             rpcAddress: atom.config.get('etheratom.rpcAddress'),
             websocketAddress: atom.config.get('etheratom.websocketAddress')
-        };
-        this._refreshSync = this._refreshSync.bind(this);
+        };        this._refreshSync = this._refreshSync.bind(this);
```
Comment (0mkara) [inline lib/components/NodeControl/index.js]: why ??? Please be careful with what you are committing.

File: lib/components/NodeControl/index.js [hunk 9]
```diff
@@ -68,13 +76,86 @@ class NodeControl extends React.Component {
...
+            toAddress: '',
+            amount: 0,
+            rpcAddress: atom.config.get('etheratom.rpcAddress'),
+            websocketAddress: atom.config.get('etheratom.websocketAddress')
+        };
+        
+        this.setState(newState);
+        this.helpers.updateWeb3();
+        try {
+
```
Comment (0mkara) [inline lib/components/NodeControl/index.js]: why newline ?

File: lib/components/NodeControl/index.js [hunk 10]
```diff
@@ -68,13 +76,86 @@ class NodeControl extends React.Component {
...
+        this.setState(newState);
+        this.helpers.updateWeb3();
+        try {
+            
+            const accounts = await this.helpers.getAccounts(); 
+            this.helpers.showPanelSuccess('Connection Re-established');
+            this.props.setAccounts( accounts );
+            if(accounts.length > 0){
+                this.props.setCoinbase(accounts[0]);
+            }else {
```
Comment (0mkara) [inline lib/components/NodeControl/index.js]: why no space ?

File: lib/components/NodeControl/index.js [hunk 11]
```diff
@@ -68,13 +76,86 @@ class NodeControl extends React.Component {
...
+            toAddress: '',
+            amount: 0,
+            rpcAddress: atom.config.get('etheratom.rpcAddress'),
+            websocketAddress: atom.config.get('etheratom.websocketAddress')
+        };
+        
+        this.setState(newState);
+        this.helpers.updateWeb3();
+        try {
+
```
Comment (0mkara) [inline lib/components/NodeControl/index.js]: extra newline

File: lib/components/NodeControl/index.js [hunk 12]
```diff
@@ -68,13 +76,86 @@ class NodeControl extends React.Component {
...
+        this.setState(newState);
+        this.helpers.updateWeb3();
+        try {
+            
+            const accounts = await this.helpers.getAccounts();            
+            this.helpers.showPanelSuccess('Connection Re-established with rpc');
+            this.props.setAccounts( accounts );
+            if(accounts.length > 0){
+                this.props.setCoinbase(accounts[0]);
+            }else {
```
Comment (0mkara) [inline lib/components/NodeControl/index.js]: no space

File: lib/components/NodeControl/index.js [hunk 13]
```diff
@@ -68,13 +76,86 @@ class NodeControl extends React.Component {
...
+                this.props.setCoinbase('0x00');
+            }
+
```
Comment (0mkara) [inline lib/components/NodeControl/index.js]: extra newline

File: lib/components/NodeControl/index.js [hunk 14]
```diff
@@ -68,13 +76,86 @@ class NodeControl extends React.Component {
...
+                this.props.setCoinbase('0x00');
+            }
+            
+        }
+        catch(e){
+            this.helpers.showPanelError('Error with RPC value. Please check again');
+            this.props.setAccounts([]);
+            this.props.setCoinbase('0x00');
+            this.props.setMining(false);
+
```
Comment (0mkara) [inline lib/components/NodeControl/index.js]: extra newline

File: index.js (hunk 1 above)
Comment (avirup2512) [reply index.js]: But due to not having semicolon..in travis build was getting failed.

Comment (0mkara) [reply index.js]: No. It was generating warnings all the time.

Comment (avirup2512) [reply index.js]: ok, revrting the semicolon.

**Pull request merged**